### PR TITLE
feat: setup the Catalog Command

### PR DESCRIPTION
**Pull request by y-lakhdar**
## Proposed changes

Setting command for catalog creation.
Parsing the data and creating the catalog is the first step in the commerce recipe.

### CODE REVIEW

File: packages/cli/src/commands/commerce/catalog/create.ts [hunk 1]
```diff
@@ -0,0 +1,139 @@
...
+import {Trackable} from '../../../lib/decorators/preconditions/trackable';
+import {AuthenticatedClient} from '../../../lib/platform/authenticatedClient';
+
+export default class CatalogCreate extends Command {
+  public static description = `${bold.bgYellow(
+    '(alpha)'
+  )} Create a commerce catalog interactively`;
+
+  public static flags = {
+    // TODO: find an alternative to skip all the prompts without having too many flags.
```
Comment (louis-bompart) [inline packages/cli/src/commands/commerce/catalog/create.ts:24]: Consider this as a UI command, i.e. we don't expect nor will support machine-user for this one

File: packages/cli/src/commands/commerce/catalog/create.ts [hunk 2]
```diff
@@ -0,0 +1,139 @@
...
+      helpGroup: 'Catalog structure',
+      description: 'Whether your catalog contains product variants',
+    }),
+    withAvailabilites: Flags.boolean({
+      char: 'a',
+      default: false,
+      helpGroup: 'Catalog structure',
+      description:
+        'Whether your catalog contains product location and/or availabilities',
+    }),
```
Comment (louis-bompart) [inline packages/cli/src/commands/commerce/catalog/create.ts:37]: can't we instead just do 'if there is at least one availability flag' ?

**Review by olamothe — APPROVED** [review]

File: packages/cli/src/commands/commerce/catalog/create.ts (hunk 2 above)
Comment (y-lakhdar) [reply packages/cli/src/commands/commerce/catalog/create.ts:37]: not sure I get your proposition. How would you do that?

Comment (louis-bompart) [reply packages/cli/src/commands/commerce/catalog/create.ts:37]: `flags.availabilityFiles.length>0`

Comment (y-lakhdar) [reply packages/cli/src/commands/commerce/catalog/create.ts:37]: Will address that in CDX-977

**Pull request merged**